### PR TITLE
[VTA][Chisel] add scalafmt and format existing scala codebase

### PR DESCRIPTION
**Pull request by vegaluisjose**
Hey @tmoreau89 @liangfu ,

This PR adds `scalafmt` plugin to both VTA and TSIM example codebase. Also, I ran formatting on the entire `scala` codebase.

### CODE REVIEW

File: vta/apps/tsim_example/hardware/chisel/.scalafmt.conf [hunk 1]
```diff
@@ -0,0 +1,2 @@
+maxColumn = 80
```
Comment (liangfu) [inline vta/apps/tsim_example/hardware/chisel/.scalafmt.conf]: I would suggest setting this number to `100`, see `max-line-length` definition in the current linter [here](https://github.com/dmlc/tvm/blob/master/tests/lint/pylintrc#L101).

File: vta/hardware/chisel/src/main/scala/core/ISA.scala [hunk 2]
```diff
@@ -79,15 +78,37 @@ trait ISAConstants
...
-  def VMIN = BitPat("b_????????_????????_??00????_????????_????????_????????_????????_????????_????????_????????_????????_????????_????????_????????_????????_?????100")
-  def VMAX = BitPat("b_????????_????????_??01????_????????_????????_????????_????????_????????_????????_????????_????????_????????_????????_????????_????????_?????100")
-  def VADD = BitPat("b_????????_????????_??10????_????????_????????_????????_????????_????????_????????_????????_????????_????????_????????_????????_????????_?????100")
-  def VSHX = BitPat("b_????????_????????_??11????_????????_????????_????????_????????_????????_????????_????????_????????_????????_????????_????????_????????_?????100")
-  def FNSH = BitPat("b_????????_????????_????????_????????_????????_????????_????????_????????_????????_????????_????????_????????_????????_????????_????????_?????011")
+  def LUOP =
+    BitPat(
+      "b_????????_????????_????????_????????_????????_????????_????????_????????_????????_????????_????????_????????_????????_????????_???????0_0????000")
```
Comment (liangfu) [inline vta/hardware/chisel/src/main/scala/core/ISA.scala]: Can we break the string into multiple segments like the following?
``` scala
def LUOP = BitPat("b_????????_????????_????????_????????_????????_????????_????????_????????" +
                   "_????????_????????_????????_????????_????????_????????_???????0_0????000")
```

File: vta/apps/tsim_example/hardware/chisel/.scalafmt.conf (hunk 1 above)
Comment (vegaluisjose) [reply vta/apps/tsim_example/hardware/chisel/.scalafmt.conf]: I just changed to 100

File: vta/hardware/chisel/src/main/scala/core/ISA.scala (hunk 2 above)
Comment (vegaluisjose) [reply vta/hardware/chisel/src/main/scala/core/ISA.scala]: I am going to fix the ISA specification on a follow up PR. That will make it easier to read.

**Pull request merged**